src/traces/box/cross_trace_calc.js

```diff
@@ -77,6 +77,11 @@ function setPositionOffset(traceType, gd, boxList, posAxis, pad) {
     var boxdv = Lib.distinctVals(pointList);
     var dPos = boxdv.minDiff / 2;
 
+    // override dPos if violin width given
+    if (vwidth != false) {
```

Comment:
🐄 I mentioned just using `if(width)` already, and anyway this block will move... but never use `==` or `!=`. Always use `===` or `!==`.

Reply:
Is this just a js convention? Do the alternatives work the same way?

Reply:
JavaScript has permission to do evil casting with `==` but not with `===` 🙂 

<img width="215" alt="screen shot 2018-10-16 at 6 34 57 pm" src="https://user-images.githubusercontent.com/15064365/47051433-46da7900-d172-11e8-8b26-fadee4b1c591.png">

Reply:
> The advice given to JavaScript beginners is to completely forget about == and to always use ===. It turns out that that rule is universally true.

http://2ality.com/2011/12/strict-equality-exemptions.html